src/replication.c

```diff
@@ -361,7 +364,7 @@ remove_replica_from_list(spec_t *spec, int iofd) {
 					MTX_LOCK(&spec->luworker_rmutex[luworker_id]);
 					pthread_cond_signal(&spec->luworker_rcond[luworker_id]); 
 					MTX_UNLOCK(&spec->luworker_rmutex[luworker_id]);
-				}else if(rcommq_ptr->acks_recvd == spec->replica_count - 1) {
+				}else if(rcommq_ptr->acks_recvd == rcommq_ptr->copies_sent - 1) {
```

Comment:
we need to do this only if acks_recvd is not met consistency_factor.. same case for below one also

Reply:
check whether we need to unset bit in 'bitset' while removing from waitq / blockedq